src/application/models/v2/Providers_model_v2.php

```diff
@@ -743,4 +743,51 @@ public function getCategoryByProvider($provider_id) {
         ->join('integrated_provider_categories', 'integrated_provider_categories.id_categories = integrated_categories.id', 'inner')
         ->where('integrated_provider_categories.id_providers', $provider_id)->get()->result_array();;
     }
+
+    public function getProvidersByIdIntegrated($providers){
+        return $this->db->select('*')->from('ea_users')->where_in('id_integrated',$providers)->get()->result_array();
+    }
+    public function addProviderToService($service_id, $providers){
+        if ( ! is_array($providers))
+        {
+            throw new Exception('Invalid argument type providers: ' . $providers);
+        }
+        if ( ! is_numeric($service_id))
+        {
+            throw new Exception('Invalid argument type idService: ' . $service_id);
+        }
+        $result = [];
+        foreach ($providers as $provider)
+        {
+            $service_provider = [
+                'id_users' => $provider['id'],
+                'id_services' => $service_id
+            ];
+            $this->db->delete('ea_services_providers', $service_provider); // this line to avoid duplicate connection when link 2 type
```

Comment:
if duplicate happens, the request should be failed as bad request, please change